### PR TITLE
Manifest fixes

### LINKED ISSUE #255: Add support for Icons using Data URIs
Add support for Icons using Data URIs, for example:
https://mobile.twitter.com/manifest.json

Uses:
"icons":[{"src":"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAA....

### PR DESCRIPTION
**Pull request by juliantoledo**
Fixes #255 and #231

### CODE REVIEW

**Comment by juliantoledo** [comment]
@andreban @ithinkihaveacat PTAL

File: lib/images.js [hunk 1]
```diff
@@ -18,61 +18,75 @@
...
 function fetchAndSave(url, destFile) {
-  return new Promise((resolve, reject) => {
-    fetch(url)
-      .then(response => {
-        if (response.status !== 200) {
-          reject(
-            new Error('Bad Response (' + response.status + ') loading image: ' + response.url));
-        }
-        const contentType = response.headers.get('Content-Type');
+  if (url.startsWith('data:')) {
```
Comment (andreban) [inline lib/images.js]: So, we currently support `http`, `https` and `data` protocol. We could write this as  
switch(protocol) {
  case 'data': {
    //...
  } 
  case 'http': 
  case 'https': {
    // ...
  }
  default: {
    return Promise.reject('Unsupported Protocol: ' + protocol);
  }
}

File: lib/images.js [hunk 2]
```diff
@@ -18,61 +18,75 @@
...
+    return this.dataUriAndSave(url);
+  }
+  return dataFetcher.fetchWithUA(url)
+    .then(response => {
+      if (response.status !== 200) {
+        return Promise.reject(new Error(
+          'Bad Response (' + response.status + ') loading image: ' + response.url));
+      }
+      const contentType = response.headers['Content-Type'];
+      return Promise.all([
```
Comment (andreban) [inline lib/images.js]: We ended up repeating the Promise.all here and in the function below. We could probably create a function with this and avoid having to change in 2 places if we decide to add another icon size, or change the sizes.

**Comment by juliantoledo** [comment]
@andreban fixed comments, agree with the PR (I got carried away on the same one).

**Pull request merged**